platform_registry_api/api.py

```diff
@@ -127,13 +127,17 @@ def _parse(cls, url: URL) -> tuple[str, str, URL]:
             mounted_repo = path_suffix.query["from"]
         return match.group("repo"), mounted_repo, path_suffix
 
-    def with_project(self, project: str) -> "RepoURL":
+    def with_project(
```

Comment:
let's rename this to `with_upstream_repo`, since we are modifying upstream image repo here. wdyt?